util/exercise/src/cmd/configure.rs

```diff
@@ -29,22 +29,28 @@ fn get_user_config(exercise_name: &str, config_content: &Value) -> Result<Value>
         Uuid::new_v4().to_hyphenated().to_string()
     };
 
-    let core = false;
-
-    let unlocked_by = loop {
+    let unlocked_by: Option<String> = loop {
         let default_value = if let Some(existing_config) = existing_config {
-            get!(existing_config, "unlocked_by", as_str)
+            get!(existing_config, "unlocked_by")
+                .as_str()
+                .map(|s| s.to_string())
```

Comment:
Instead of converting `default_value` to `Option<String>`, which requires a bunch of expansions below to do the right thing, what about restructuring the code like this:

```rust
let default_value = if let Some(existing_config) = existing_config {
    if let Some(Value::String(ref s)) = get!(existing_config, "unlocked_by") {
        s
    } else {
        "none"
    }
} else {
    "hello-world"
}
```

I haven't run that snippet past the compiler, so there may need to be some small changes, but I think what you end up with will be more readable and more usable than keeping this current setup.